spring-5-security/src/main/java/com/baeldung/xss/XSSUtils.java

```diff
@@ -0,0 +1,20 @@
+package com.baeldung.xss;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import org.owasp.esapi.ESAPI;
+
+public class XSSUtils {
+
+    public static String stripXSS(String value) {
+        if (value == null) {
+            return null;
+        }
+        value = ESAPI.encoder()
+            .canonicalize(value);
+        value = value.replaceAll("\0", "");
+        value = Jsoup.clean(value, Whitelist.none());
+        return value;
```

Comment:
It's generally bad form to keep redefining a variable - especially one which is also an input parameter to a function. Can we rewrite?

E.g.

```java
// use fluency to reduce need for intermediate variables
String normalized = ESAPI.encoder()
  .canonicalize(value)
  .replaceAll("\0", "");
  
// use a return rather than assign-then-return
return Jsoup.clean(normalized, Whitelist.none());
```

Reply:
Done.